app/controllers/locations_controller.rb

```diff
@@ -1,25 +1,29 @@
 class LocationsController < ApplicationController
   def create
-    @location = build_location
+    @profile = current_user
+
+    if save_location?
+      enqueue_geocode_job
 
-    if update_location?
       redirect_to profile_url
     else
-      @profile = current_user
-
       render "profiles/show"
```

Comment:
Was no change required in the template? Or is `profiles/show` still using the `@location` directly?